UnityProject/Assets/Scripts/Objects/Doors/ConstructibleDoor.cs

```diff
@@ -1,56 +1,120 @@
+		private GameObject airlockElectronicsPrefab = null;
+
+		public GameObject AirlockElectronicsPrefab => airlockElectronicsPrefab;
+
 		public bool Reinforced = false;
 
 		private bool panelopen = false;
 
 		public bool Panelopen => panelopen;
 
+		private DoorMasterController doorMasterController;
+		private BoltsModule boltsModule;
+		private WeldModule weldModule;
+		private Integrity integrity;
+
+		private void Awake()
+		{
+			doorMasterController = GetComponent<DoorMasterController>();
+			boltsModule = GetComponentInChildren<BoltsModule>();
+			weldModule = GetComponentInChildren<WeldModule>();
+
+			if (CustomNetworkManager.IsServer == false) return;
+
+			integrity = GetComponent<Integrity>();
+			integrity.OnWillDestroyServer.AddListener(WhenDestroyed);
+
+		}
+
 		public bool WillInteract(HandApply interaction, NetworkSide side)
 		{
-			if (DefaultWillInteract.Default(interaction, side) == false) return false;
-			if (Validations.HasItemTrait(interaction.UsedObject, CommonTraits.Instance.Screwdriver) == false)
+			if (!DefaultWillInteract.Default(interaction, side) || interaction.TargetObject != gameObject)
 				return false;
-			return true;
+
+			if (Validations.HasItemTrait(interaction.UsedObject, CommonTraits.Instance.Screwdriver))
+				return true;
+
+			if (!weldModule.CanDoorStateChange() && boltsModule.CanDoorStateChange() && !doorMasterController.HasPower)
+			{
+				return Validations.HasItemTrait(interaction.UsedObject, CommonTraits.Instance.Crowbar);
+			}
+
+			return false;
 		}
 
 		public void ServerPerformInteraction(HandApply interaction)
 		{
-			panelopen = !panelopen;
-			if (panelopen)
+			if (Validations.HasItemTrait(interaction.UsedObject, CommonTraits.Instance.Screwdriver))
 			{
-				DoorAnimatorV2.AddPanelOverlay();
-				Chat.AddActionMsgToChat(interaction.Performer,
-					$"You unscrews the {gameObject.ExpensiveName()}'s cable panel.",
-					$"{interaction.Performer.ExpensiveName()} unscrews {gameObject.ExpensiveName()}'s cable panel.");
+				panelopen = !panelopen;
+				if (panelopen)
+				{
+					DoorAnimatorV2.AddPanelOverlay();
+					Chat.AddActionMsgToChat(interaction.Performer,
+						$"You unscrews the {gameObject.ExpensiveName()}'s cable panel.",
+						$"{interaction.Performer.ExpensiveName()} unscrews {gameObject.ExpensiveName()}'s cable panel.");
+				}
+				else
+				{
+					DoorAnimatorV2.RemovePanelOverlay();
+					Chat.AddActionMsgToChat(interaction.Performer,
+						$"You screw in the {gameObject.ExpensiveName()}'s cable panel.",
+						$"{interaction.Performer.ExpensiveName()} screws in {gameObject.ExpensiveName()}'s cable panel.");
+
+					//Force close net tab when panel is closed
+					TabUpdateMessage.SendToPeepers(gameObject, NetTabType.HackingPanel, TabAction.Close);
+				}
+
+
+				AudioSourceParameters audioSourceParameters =
+					new AudioSourceParameters(pitch: UnityEngine.Random.Range(0.8f, 1.2f));
+				SoundManager.PlayNetworkedAtPos(CommonSounds.Instance.screwdriver,
```

Comment:
The doc string recommends that you use the async method of this
don't forget to add `_ =` if you choose to use that method

Reply:
```suggestion
				_ = SoundManager.PlayNetworkedAtPos(CommonSounds.Instance.screwdriver,
```
Do you mean that? But this method returns nothing. I don't quite understand why I need to use discard here. Actually, this code was already here, I did not write it

Reply:
You can discard the result of the async method - I think it is `PlayNetworkedAtPosAsync`?